product/stream_processor/integrations/events/providers/eventbridge.py

```diff
@@ -0,0 +1,87 @@
+import os
+from typing import TYPE_CHECKING, Generator, Optional
+
+import boto3
+import botocore.exceptions
+
+from product.constants import XRAY_TRACE_ID_ENV
+from product.stream_processor.integrations.events.base import EventProvider
+from product.stream_processor.integrations.events.constants import EVENTBRIDGE_PROVIDER_MAX_EVENTS_ENTRY
+from product.stream_processor.integrations.events.exceptions import ProductNotificationDeliveryError
+from product.stream_processor.integrations.events.functions import chunk_from_list
+from product.stream_processor.integrations.events.models.input import Event
+from product.stream_processor.integrations.events.models.output import EventReceipt, EventReceiptFail, EventReceiptSuccess
+
+if TYPE_CHECKING:
+    from mypy_boto3_events import EventBridgeClient
+    from mypy_boto3_events.type_defs import PutEventsRequestEntryTypeDef, PutEventsResponseTypeDef
+
+
+class EventBridge(EventProvider):
+
+    def __init__(self, bus_name: str, client: Optional['EventBridgeClient'] = None):
+        self.bus_name = bus_name
+        self.client = client or boto3.client('events')
+
+    def send(self, payload: list[Event]) -> EventReceipt:
+        success: list[EventReceiptSuccess] = []
+        failed: list[EventReceiptFail] = []
+        events = self.build_put_events_requests(payload)
+
+        for batch in events:
+            try:
+                result = self.client.put_events(Entries=batch)
+                ok, not_ok = self._collect_receipts(result)
+                success.extend(ok)
+                failed.extend(not_ok)
+            except botocore.exceptions.ClientError as exc:
+                error_message = exc.response['Error']['Message']
+
+                receipt = EventReceiptFail(receipt_id='', error='error_message', details=exc.response['ResponseMetadata'])
+                raise ProductNotificationDeliveryError(f'Failed to deliver all events: {error_message}', receipts=[receipt]) from exc
```

Comment:
about the error handling here, if we raise the exception, we stop going over the batch right? shouldnt we do failed.extend(batch somehow) and do continue in the loop?